src/Main.java

```diff
@@ -1,42 +1,106 @@
+import java.awt.desktop.PreferencesEvent;
 import java.util.HashSet;
 import java.util.Objects;
+import java.util.Random;
 import java.util.Scanner;
 
 public class Main {
     public static void main(String[] args) {
 
         // Hier mag je je code scrijven voor de hoofd-opdracht
+        Integer[] numeric = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        String[] alphabetic = {"nul", "één", "twee", "drie", "vier", "vijf", "zes", "zeven", "acht", "negen"};
 
-        /* deze regel mag je weg halen voor de bonus opdracht. Onderstaande code is voor de bonus opdracht.
-        HashSet<Integer> secretnumber = randomnumbergenerator();
-        String stringnumber =  setToStringConverter(secretnumber);
-        System.out.println(stringnumber);
-        feedback();
-         deze regel mag je weg halen voor de bonus opdracht */
+        Translator translator = new Translator(alphabetic, numeric);
+
+        boolean play = true;
+        String ongeldig = "Ongeldige invoer";
+        Scanner scanner = new Scanner(System.in);
+
+        while (play) {
+            System.out.println("Type 'x' om te stoppen \nType 'v' om te vertalen");
+            String input = scanner.nextLine();
+
+            if (input.equals("x")) {
+                play = false;
+            } else if (input.equals("v")) {
+                System.out.println("Type een cijfer van 0 t/m 9");
+                int number = scanner.nextInt();
+                scanner.nextLine();
+                if (number < 10 && number >= 0) {
+                    String result = translator.translate(number);
+                    System.out.println("De vertaling van " + number + " is " + result);
+                } else {
+                    System.out.println(ongeldig);
+                }
+            } else {
+                System.out.println(ongeldig);
+            }
+        }
+
+
+        HashSet<Integer> secretNumber = randomNumberGenerator();
+        String stringNumber =  setToStringConverter(secretNumber);
+        System.out.println(stringNumber);
+        feedback(stringNumber);
 
     }
 
+    public static HashSet<Integer> randomNumberGenerator() {
+        HashSet<Integer> randomNumbers = new HashSet<>();
+        Random random = new Random();
+        int i = 0;
```

Comment:
i had weggelaten kunnen worden en vervangen worden door `randomNumber.size` . 
Vervolgens had in de while-loop onderstaande moeten staan. Nu voeg hij een cijferreeks van 4 en soms 5 cijfers toe.
int x = random.nextInt(10);
             randomNumber.add(x);